R/helpers.R

```diff
@@ -356,36 +356,41 @@ printVignette <- function (x, ...)
     invisible(x)
 }
 
+# Validate-class ----
 
-##' A Reference Class to help programming validation for new S4 classes
-##' 
-##'
-##' @description `r lifecycle::badge("stable")`
-##'
-##' Starting from an empty \code{msg} vector, with each check that is returning
-##' FALSE the vector gets a new element - the string explaining the failure of
-##' the validation
-##'
-##' @name Validate
-##' @field msg the message character vector
-Validate <-
-    setRefClass("Validate",
-                fields =
-                list(msg = "character"),
-                methods = list(
-                check =
-                    function(test,
-                             string){
-                        if(test)
-                        {} else {
-                            msg <<- c(msg,
-                                      string)
-                        }
-                    },
-                result =
-                    function() {
-                        if(length(msg) > 0) msg else TRUE
-                    }))
+#' `Validate`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' The [`Validate`] class is a Reference Class
+#' to help programming validation for new S4 classes.
+#'
+#' @details Starting from an empty \code{msg} vector, with each check
+#' that is returning `FALSE` the vector gets a new element - the string
+#' explaining the failure of the validation.
+#'
+#' @name Validate
+#' @field msg the message character vector
+Validate <- setRefClass(
+  Class = "Validate",
+  fields = list(msg = "character"),
+  methods = list(
+    check = function(test, string) {
+      if (test) {
+        NULL
+      } else {
+        msg <<- c(msg, string)
```

Comment:
since we use here the `<<-` assignment operator, can you check in a dedicated test that having a `msg` variable in the test does not confuse this method?

Reply:
@danielinteractive - Do you mean dedicate unit test? If it so, sure, I will create something I believe is ok - but please double check it after I commit it, since I am not so sure if it will be ok. Also I think there should not be any confusions since, the msg ("global") variable stays enclosed in the environment of the instance of `Variable` class.

Reply:
yeah dedicated test. agree it should be ok but good to be sure.

Reply:
Absolutely, I added 4 quite different tests - please decide which one of them should stay and which should be removed. I hope there are differ enough so that they can stay all :):).